### ISSUE TITLE
@ApiRequestParam

### ISSUE THREAD

**Issue by pereng** [fabiomaffioletti/jsondoc#85]
Hi!

In Spring it is possible to annotate a variable in a method with org.springframework.web.bind.annotation.RequestHeader indicating: "that a method parameter should be bound to a web request header" (according to Spring JavaDoc). Would it be possible to extend JSONDoc with a new annotation to document this type of parameter. For instance: @ApiRequestParam?

Our current use case is that the customer needs to set a custom HTTP Request Param used for authentication. It would be really nice to be able to document this in the API documentation.

Regards,
Per

**Comment by matt-blanchette** [comment]
I have a slightly similar use case for a auth header.
I am looking at using the @ApiHeaders type. This can be specified at the method level (though my use case would be easier at the class level).

But it sounds like you would like the SpringJSONDocScanner.getHeadersFromSpringAnnotation to handle RequestHeader similar to how RequestParam is handled.

**Comment by fabiomaffioletti** [comment]
Will be fixed in version 1.1.10, to be released shortly. @RequestHeader annotation will still be activated by the @ApiHeader on the same parameter, for example:

```
public Book header(@RequestHeader(value="myHeader") @ApiHeader String myHeader) {
//...
}
```
If you don't specify @ApiHeader on the same parameter, then the item is not documented.

**Comment by pereng** [comment]
Sweet! Works like a charm ;) Thank you
